bundle/jsky.app.ot/src/main/scala/jsky/app/ot/editor/seq/ItcTable.scala

```diff
@@ -128,6 +179,62 @@ trait ItcTable extends Table {
     o.flatten.fold("Could not identify ags strategy or guide probe type".left[GuideProbe])(_.right)
   }
 
+  private def extractSource(target: ITarget, c: ItcUniqueConfig): String \/ SourceDefinition = {
+    for {
+      (mag, band) <- extractSourceMagnitude(target, c.config)
+    } yield {
+      // TODO: definition of spectral profile and redshift
+      new SourceDefinition(PointSource(mag, BrightnessUnit.MAG), /*TODO*/LibraryStar("A0V"), band, /*TODO*/0.0)
+    }
+  }
+
+  private def extractSourceMagnitude(target: ITarget, c: Config): String \/ (Double, WavebandDefinition) = {
+
+    def closestBand(bands: List[Magnitude], wl: Double) =
+      bands.minBy(m => Math.abs(m.getBand.getWavelengthMidPoint.getValue.toDouble - wl))
+
+    def mags(wl: Double): String \/ Magnitude = {
+      val bands = target.getMagnitudes.toList.asScala.toList.
+        filter(_.getBand.getWavelengthMidPoint.isDefined).// ignore bands with unknown wavelengths (currently AP only)
+        filterNot(_.getBand == Magnitude.Band.UC).        // ignore UC magnitudes
+        filterNot(_.getBand == Magnitude.Band.AP)         // ignore AP magnitudes
+      if (bands.isEmpty) "No standard magnitudes for target defined; ITC can not use UC and AP magnitudes.".left[Magnitude]
+      else closestBand(bands, wl).right[String]
+    }
+
+    for {
+      wl  <- ConfigExtractor.extractObservingWavelength(c)
+      mag <- mags(wl*1000) // convert wavelength from micrometer [um] to nanometer [nm]
+    } yield {
+      val b = mag.getBrightness
+      mag.getBand match {
+        // TODO: unify band definitions from spModel core and itc shared so that we don't need this translation anymore
+        case Magnitude.Band.u  => (b, WavebandDefinition.U)
```

Comment:
Oh. We actually have 3 band definitions now: new and old from core model and one from ITC. Ok, I'll clean that up when I replace the ITC bands with the new core model ones... watch out for OCSADV-305.